app/assets/javascripts/discourse/app/services/header.js

```diff
@@ -29,4 +33,22 @@ export default class Header extends Service {
       }
     }
   }
+
+  registerHider(ref, buttons) {
```

Comment:
I will make a validator in a followup PR so only `search`, `signup`, and `login` buttons are allowed here, maybe extract to consts as well.